typing/typemod.ml

```diff
@@ -1816,85 +1945,116 @@ let type_package env m p nl =
   (wrap_constraint env true modl mty Tmodtype_implicit, tl')
 
 (* Fill in the forward declarations *)
+
+let type_open ?used_slot ovf env loc me =
+  type_open_ ?used_slot ?toplevel:None ovf env loc me
+
+let gen_mod_ident me =
+  match me.pmod_desc with
+  | Pmod_functor _ | Pmod_extension _ -> assert false
```

Comment:
You should not `assert false` here, but instead report a proper error.  Otherwise:

````ocaml
 # let f () = let open functor(X: sig end) -> struct end in ();;
Fatal error: exception File "typing/typemod.ml", line 1954, characters 41-47: Assertion failed
# let f () = let open [%foo] in ();;
Fatal error: exception File "typing/typemod.ml", line 1954, characters 41-47: Assertion failed
````

Reply:
done